kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisSchedulerSource.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.kinesis.scaladsl
+
+import akka.stream.Supervision.{Resume, Stop}
+import akka.stream._
+import akka.stream.alpakka.kinesis.impl.KinesisSchedulerSourceStage
+import akka.stream.alpakka.kinesis.{
+  CommittableRecord,
+  KinesisSchedulerCheckpointSettings,
+  KinesisSchedulerSourceSettings
+}
+import akka.stream.scaladsl.{Flow, GraphDSL, Sink, Source, Zip}
+import akka.{Done, NotUsed}
+import software.amazon.kinesis.coordinator.Scheduler
+import software.amazon.kinesis.exceptions.ShutdownException
+import software.amazon.kinesis.processor.ShardRecordProcessorFactory
+import software.amazon.kinesis.retrieval.KinesisClientRecord
+
+import scala.collection.immutable
+import scala.concurrent.Future
+
+object KinesisSchedulerSource {
+
+  /**
+   * Source that spawns a single Kinesis KCL Scheduler capable of consuming records from multiple Kinesis shards in coordination with other KCL Scheduler instances connected to the same stream. Only a single KCL Scheduler instance can own the lease of a shard at any given time. The Kinesis service assigns each shard lease among the available instances.
+   *
+   * Every message emitted downstream belongs to a shard. Messages can be committed to save the progress of the client consuming that shard.
+   *
+   * @param schedulerBuilder
+   * @param settings
+   * @return
+   */
+  def apply(
+      schedulerBuilder: ShardRecordProcessorFactory => Scheduler,
+      settings: KinesisSchedulerSourceSettings = KinesisSchedulerSourceSettings.defaultInstance
+  ): Source[CommittableRecord, NotUsed] =
+    Source
+      .setup { (mat, _) =>
+        import mat.executionContext
+        Source
+          .fromGraph(new KinesisSchedulerSourceStage(settings, schedulerBuilder))
+      }
+      .mapMaterializedValue(_ => NotUsed)
```

Comment:
This will do the trick to retain the Scheduler future.
```scala
      .mapMaterializedValue(_.flatMap(identity)(ExecutionContexts.sameThreadExecutionContext))
```